Add documentation of the vulnerability detector tests (#820)
wazuh/wazuh-qa: tests/integration/test_vulnerability_detector/test_scan_results/readme.md

# Test scan results

## Objective

The purpose of testing providers settings is to check if the vulnerability detector generates alerts in the
appropriate cases.

## General info

|Tier | Number of tests | Time spent |
|:--:|:--:|:--:|
| 0 | 34 | 0:12:34 |

## Expected behavior

After inserting custom vulnerabilities and vulnerable packages, vulnerability detector has to generate vulnerability
alerts. Some considerations to take into account:

- It is necessary that NVD is enabled in `ossec.conf` for scanning vulnerability alerts.
- At least one vulnerability needs to be inserted in the NVD and `vulnerabilities` tables to generate vulnerability
alerts.

## Testing

### Generic tests

> - **[test_debian_inventory_debian_feed](../doc/test_scan_results/test_debian_inventory_debian_feed.md)**: Tests that
> are based on mocking debian systems and inserting custom vulnerabilities and vulnerable packages to see if vulnerability
> detector generates the vulnerability alerts from Debian provider feed.
> 
> - **[test_redhat_inventory_redhat_feed](../doc/test_scan_results/test_redhat_inventory_redhat_feed.md)**: Tests that are
> based on mocking redhat systems and inserting custom vulnerabilities and vulnerable packages to see if vulnerability
> detector generates the vulnerability alerts from Red Hat provider feed.
> 
> - **[test_ubuntu_inventory_canonical_feed](../doc/test_scan_results/test_ubuntu_inventory_canonical_feed.md)**: Tests
> that are based on mocking ubuntu systems and inserting custom vulnerabilities and vulnerable packages to see if
> vulnerability detector generates the vulnerability alerts from Canonical provider feed.
> 
> - **[test_scan_nvd_feed](../doc/test_scan_results/test_scan_nvd_feed.md)**: Tests that are based on mocking Red Hat,
> Canonical, Debian and Windows systems, and inserting custom vulnerabilities and vulnerable packages to see if
> vulnerability detector generates the vulnerability alerts from NVD feed.
> 
> - **[test_scan_providers_and_nvd_feed](../doc/test_scan_results/test_scan_providers_and_nvd_feed.md)**: Tests that are
> based on mocking Red Hat, Canonical and Debian systems, and inserting custom vulnerabilities and vulnerable packages to
> see if vulnerability detector generates the vulnerability alerts from NVD and providers feed.

Reviewer: ```suggestion
- **[test_debian_inventory_debian_feed](../doc/test_scan_results/test_debian_inventory_debian_feed.md)**: Tests that mock Debian systems and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from Debian provider feed.

- **[test_redhat_inventory_redhat_feed](../doc/test_scan_results/test_redhat_inventory_redhat_feed.md)**: Tests that mock RedHat systems and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from RedHat provider feed.

- **[test_ubuntu_inventory_canonical_feed](../doc/test_scan_results/test_ubuntu_inventory_canonical_feed.md)**: Tests
that mock Ubuntu systems and insert custom vulnerabilities and vulnerable packages to see if Vulnerability Detector generates alerts from Canonical provider feed.

- **[test_scan_nvd_feed](../doc/test_scan_results/test_scan_nvd_feed.md)**: Tests that mock RedHat, Ubuntu, Debian and Windows systems, and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from NVD feed.

- **[test_scan_providers_and_nvd_feed](../doc/test_scan_results/test_scan_providers_and_nvd_feed.md)**: Tests that mock RedHat, Ubuntu and Debian systems, and insert custom vulnerabilities and vulnerable packages to check if Vulnerability Detector generates alerts from NVD and providers feed.
```
